tests/Integrations/PDO/PDOSandboxedTest.php

```diff
@@ -0,0 +1,398 @@
+            )
+                ->setTraceAnalyticsCandidate()
+                ->withExactTags(array_merge($this->baseTags(), [
+                    'db.rowcount' => '1',
+                ])),
+            SpanAssertion::build(
+                'PDO.prepare',
+                'PDO',
+                'sql',
+                "SELECT * FROM tests WHERE id = ?"
+            )->withExactTags(array_merge($this->baseTags(), [])),
+            SpanAssertion::exists('PDO.__construct'),
+        ]);
+    }
+
+    public function testPDOStatementError()
+    {
+        $query = "WRONG QUERY";
+        $traces = $this->isolateTracer(function () use ($query) {
+            try {
+                $pdo = $this->pdoInstance();
+                $stmt = $pdo->prepare($query);
+                $stmt->execute([1]);
+                $stmt->fetchAll();
+                $stmt->closeCursor();
+                $stmt = null;
+                $pdo = null;
+            } catch (\PDOException $ex) {
+            }
+        });
+        $this->assertSpans($traces, [
+            SpanAssertion::build('PDOStatement.execute', 'PDO', 'sql', "WRONG QUERY")
+                ->setTraceAnalyticsCandidate()
+                ->setError('PDOStatement error', 'SQL error: 42000. Driver error: 1064')
+                    ->withExactTags($this->baseTags()),
+            SpanAssertion::build('PDO.prepare', 'PDO', 'sql', "WRONG QUERY")
+                ->withExactTags($this->baseTags()),
+            SpanAssertion::exists('PDO.__construct'),
+        ]);
+    }
+
+    public function testPDOStatementException()
+    {
+        $query = "WRONG QUERY";
+        $traces = $this->isolateTracer(function () use ($query) {
+            try {
+                $pdo = $this->pdoInstance();
+                $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
+                $stmt = $pdo->prepare($query);
+                $stmt->execute([1]);
+                $stmt->fetchAll();
+                $stmt->closeCursor();
+                $stmt = null;
+                $pdo = null;
+            } catch (\PDOException $ex) {
+            }
+        });
+        $this->assertSpans($traces, [
+            SpanAssertion::build('PDOStatement.execute', 'PDO', 'sql', $query)
+                ->setTraceAnalyticsCandidate()
+                // phpcs:disable
+                ->setException('PDOException', 'SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near \'WRONG QUERY\' at line 1')
+                // phpcs:enable
+                ->withExactTags($this->baseTags()),
+            SpanAssertion::build('PDO.prepare', 'PDO', 'sql', $query)
+                ->withExactTags($this->baseTags()),
+            SpanAssertion::exists('PDO.__construct'),
+        ]);
+    }
+
+    // Add limited tracer to sandboxed closures
+    /*
```

Comment:
This test / and its functionality needs to be implemented . Since PDO can create tens of thousands Spans easily, this mode ensures that we stop creating spans when configured limit is reached. In future the limit might be increased but in general it should be there to safeguard against excessive resource consumption.